From 9068bfeb89c4baf6a5b4f61cfcc10475c68dab32 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Tue, 11 Sep 2012 12:09:53 -0400
Subject: [PATCH 26/55] Fix: statedump missing check for NULL namespace

commit debfbc4ed959e6223cea1926d31255edbe51de9b lttng.org/lttng-modules.git

Cause kernel OOPS. statedump PID namespace support is newly introduced
in lttng-modules 2.1. Happens when iterating on a zombie process.

Reported-by: David Goulet <dgoulet@efficios.com>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Signed-off-by: Paul Barrette <paul.barrette@windriver.com>
---
 .../events/lttng-module/lttng-statedump.h          |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/staging/lttng/instrumentation/events/lttng-module/lttng-statedump.h b/drivers/staging/lttng/instrumentation/events/lttng-module/lttng-statedump.h
index ea0b6d9..d350cad 100644
--- a/drivers/staging/lttng/instrumentation/events/lttng-module/lttng-statedump.h
+++ b/drivers/staging/lttng/instrumentation/events/lttng-module/lttng-statedump.h
@@ -67,10 +67,12 @@ TRACE_EVENT(lttng_statedump_process_state,
 				struct task_struct *parent;
 				pid_t ret = 0;
 
-				rcu_read_lock();
-				parent = rcu_dereference(p->real_parent);
-				ret = task_tgid_nr_ns(parent, pid_ns);
-				rcu_read_unlock();
+				if (pid_ns) {
+					rcu_read_lock();
+					parent = rcu_dereference(p->real_parent);
+					ret = task_tgid_nr_ns(parent, pid_ns);
+					rcu_read_unlock();
+				}
 				ret;
 			}))
 		tp_memcpy(name, p->comm, TASK_COMM_LEN)
-- 
1.7.9.7

